code/viprchk.cpp

```diff
@@ -961,11 +960,18 @@ bool processDER()
 
       int newConIdx = constraint.size();
 
+      if ( !warning_issued && parseNumber(label) != newConIdx) {
+         cerr << "Warning: indices are not ascending for " << label << ". This can indicate an error!"  << endl;
+         warning_issued = true;
+      }
```

Comment:
Not sure: e.g. in examples/paper_eg3.vipr assumption and constraint labels are counted independently.

Reply:
Also labels can have arbitrary names, since they are not used. They were originally just meant to make certificates more human-readable. (But actually that should be achieved by other means such as comments I guess.)